### PR TITLE
ref #2183 add socket factory to JedisPool

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/main/java/redis/clients/jedis/JedisFactory.java [hunk 1]
```diff
@@ -19,17 +19,13 @@
  * PoolableObjectFactory custom impl.
  */
 class JedisFactory implements PooledObjectFactory<Jedis> {
-  private final AtomicReference<HostAndPort> hostAndPort = new AtomicReference<HostAndPort>();
-  private final int connectionTimeout;
-  private final int soTimeout;
+  
+  private final JedisSocketFactory jedisSocketFactory;
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisFactory.java:23]: Shouldn't we use `AtomicReference`, may be in JedisSocketFactory, to match previous implementation?

Comment (gkorland) [reply src/main/java/redis/clients/jedis/JedisFactory.java:23]: @sazzad16 I'm not sure why it was there to begin with

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/JedisFactory.java:23]: @gkorland #681

**Comment by sazzad16** [comment]
@gkorland Close due to #2151 ?

**Pull request closed without merging**